### ISSUE TITLE
FreeViz automatically runs when changing Gravity settings

### ISSUE THREAD

**Issue by processo** [biolab/orange3#6688]
FreeViz runs automatically when Gravity controls are changed in certain situations. I'm not sure if this is intended because every other setting is applied immediately, but the the fact that the optimization is stopped would suggest it should not start until I press Start. Either way, it still behaves in two different ways.

When it happens:
1. Click the Start button in FreeViz to **run it at least once**
2. Check or uncheck Gravity, or move the slider

When it does **not** happen:
- FreeViz has not been run yet
- When Initizalization is changed

OS: Windows 10
Orange: 3.36.1

**Comment by janezd** [comment]
Dear @processo, I spent like two full weeks, reimplementing the entire vizrank base code and the five widget that use it, and all because of your issue #6504. Now that this is finally done, you waste no time filing another, similar issue related to freeviz?

Just joking. VizRank reimplementation indeed took some time, but it was worth it, so thanks for triggering it. The old behavior was indeed inconsistent, as you wrote, and the code contained too many patches and small bugs. The new one makes me proud.

You're right here, too. I like to change gravity during optimization and immediately see the result. To make it consistent, changing the gravity shouldn't restart the optimization it it's already done. I don't like this because then the user would have to catch it before it stops, or restart if he's too late.

I suppose we should just eliminate the start button? Optimization should start immediately, and restart (from current position) when the user changes initialization, gravity or color. No point in waiting with optimization. If the user wants to restart from random (or circular), he can simply (re)select the option in the Initialization combo.

Or should we still offer Pause and Resume? And when paused, changing gravity and initialization would have no effect? In other words, we'd replace Start / Pause / Resume with Pause / Resume?

**Comment by processo** [comment]
Lol. I did not want to leave you without work for too long.

On second thought though, the current behavior seems like the best option. I worked a lot on my old laptop with large tables and pulling sliders while optimization running was a pain. It is not a problem though with today's performance. About eliminating the Start button, it is still good that it does not run immediately after connecting the widget. I like to have a chance to review setting before it runs. And maybe in some cases the user wants to stop it early.

So I guess there is no work to be done afterall.

PS. I am deeply grateful for Gravity. I dug up two old document classification projects after it's implementation and found spectacularly better solutions with it.

**Comment by janezd** [comment]
We discussed this today. There seems to be no harm in running immediately because the widget is responsive while FreeViz runs. (We need to make sure: we have to check this with large data on a slow machine.) But we'd keep a kind of on/off button that would freeze/unfreeze optimization.
